code/modules/client/medal_persistence.dm

```diff
@@ -0,0 +1,267 @@
+
+	var/issued_to_real_name //! The real name of the player who was issued the medal
+	var/issued_to_rank //! The rank of the player who was issued the medal
+
+	var/issued_by_real_name //! The real name of the player who issued the medal
+	var/issued_by_rank //! The rank of the player who issued the medal
+
+	var/obj/item/clothing/tie/medal/medal //! The medal object
+	var/obj/item/clothing/tie/medal/medal_uid //! The unique id of the medal
+	var/medal_citation //! The citation on the medal
+
+	var/issued_at //! The time the medal was issued
+	var/is_posthumous //! Whether the medal was awarded posthumously
+
+/datum/persistent_medal_info/Destroy(force, ...)
+	if(medal)
+		// unregister BEFORE qdel'ing it
+		UnregisterSignal(medal, COMSIG_QDELETING)
+		QDEL_NULL(medal)
+	return ..()
+
+/**
+ * Load a medal from a data record.
+ * - list/data - The data record to load the medal from.
+ */
+/proc/load_persistent_medal_from_data(list/data)
+	RETURN_TYPE(/datum/persistent_medal_info)
+	var/datum/persistent_medal_info/medal = new
+
+	medal.id = data["id"]
+
+	medal.issued_to_real_name = data["issued_to_real_name"]
+	medal.issued_to_rank = data["issued_to_rank"]
+	medal.issued_by_real_name = data["issued_by_real_name"]
+	medal.issued_by_rank = data["issued_by_rank"]
+
+	medal.medal_uid = data["medal_uid"]
+	medal.medal_citation = data["medal_citation"]
+
+	medal.medal.recipient_name = medal.issued_to_real_name
+	medal.medal.recipient_rank = medal.issued_to_rank
+	medal.medal.medal_citation = medal.medal_citation
+
+	medal.issued_at = data["issued_at"]
+	medal.is_posthumous = data["is_posthumous"]
+
+	return medal
+
+/datum/persistent_medal_info/proc/get_metal_typepath_from_uid()
+	var/static/list/uid_map
+
+	if(isnull(uid_map))
+		uid_map = list()
+		for(var/datum/persistent_medal_info/medal_type as anything in subtypesof(/obj/item/clothing/tie/medal))
+			var/uid = medal_type::medal_uid
+			if(uid in uid_map)
+				stack_trace("duplicate medal_uid found: `[uid]` in `[medal_type]` and `[uid_map[uid]]`")
+			uid_map[uid] = medal_type
+	return uid_map[medal_uid]
+
+/**
+ * Generate the medal and put it in the provided container.
+ * - mob/container - The mob being awarded the medal.
+ */
+/datum/persistent_medal_info/proc/award_to(atom/container)
+	var/wanted_type = get_metal_typepath_from_uid()
+	if(isnull(wanted_type))
+		stack_trace("Failed to find a medal type for medal_uid `[medal_uid]`")
+		return
+
+	medal = new wanted_type
+	if(isliving(container))
+		var/mob/living/mob = container
+		mob.put_in_hands(medal)
+		return
+	medal.forceMove(container)
+	RegisterSignal(medal, COMSIG_QDELETING, PROC_REF(on_medal_destroy))
+
+/**
+ * Handle the destruction of the medal.
+ */
+/datum/persistent_medal_info/proc/on_medal_destroy()
+	SIGNAL_HANDLER
+
+	medal_persistence.medals_by_real_name[issued_to_real_name] -= src
+	ASYNC
+		medal_persistence.mark_medal_as_deleted(src)
+		to_chat(medal_persistence.owner, span_notice("A medal was destroyed..."))
```

Comment:
```suggestion
```
We dont want to have medals be deleted deleted when they get destroyed, since the point of them is to reward people doing something impressive, and in tgmc theres a lot of way to get gibbed beyond your control

We'll need a UI to let admins delete problematic medals but since I have an admin verb refactor open and I feel like doing some tgui I can do that for you after a test?